parsec/parsec.c

```diff
@@ -2759,3 +2759,51 @@ void parsec_set_my_execution_stream(parsec_execution_stream_t *es)
 {
     PARSEC_TLS_SET_SPECIFIC(parsec_tls_execution_stream, es);
 }
+
+
+/**
+ * Query various properties of a PaRSEC context.
+ */
+int parsec_context_query(parsec_context_t *context, parsec_context_query_cmd_t cmd, ...)
+{
+    parsec_device_module_t* dev;
+    va_list args;
+    va_start(args, cmd);
+
+    switch(cmd) {
+        case PARSEC_CONTEXT_QUERY_NODES:
+            switch (parsec_communication_engine_up) {
+                case 0: return 0;  /* context not ready for distributed runs, and lacking datatype chandling capabilities */
+                case 1: return 1;  /* single node runs, but the context has datatype management capabilties */
+                case 2: return -1; /* we are in a distributed run, but the MPI engine is not yet ready, so the nb_nodes might not be accurate */
```

Comment:
PARSEC_ERROR and -1 are the same numerical value. 

Maybe we could return PARSEC_ERR_NOT_FOUND in that case?